config/config.go

```diff
@@ -972,6 +974,11 @@ func (c *AlertmanagerConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 		return fmt.Errorf("at most one of basic_auth, authorization, oauth2, & sigv4 must be configured")
 	}
 
+	// Disallow config using AM V1 API.
```

Comment:
I think `AlertmanagerAPIVersion.UnmarshalYAML` should already be taking care of this.